R/trans-numeric.r

```diff
@@ -16,17 +16,56 @@ atanh_trans <- function() {
   trans_new("atanh", "atanh", "tanh")
 }
 
-#' Box-Cox (modulus) power transformation.
+#' Box-Cox transformations.
 #'
-#' @param p Exponent of Box-Cox transformation.
-#' @details Implements a generalisation of the Box-Cox transformation that works
-#' for data with both positive and negative values.
-#' @references J. John and N. Draper. "An alternative family of
-#' transformations." Applied Statistics, pages 190-197, 1980.
-#' \url{http://www.jstor.org/stable/2986305}
+#' Transformations used to transform non-normal data
+#' to more closely approximate a normal distribution.
+#'
+#' @param p Transformation exponent, \eqn{\lambda}.
+#' @details The Box-Cox power transformation requires strictly positive values and
+#' takes the following form for `y > 0`:
+#' \deqn{y^{(\lambda)} = \frac{y^\lambda - 1}{\lambda}}{y^(\lambda) = (y^\lambda - 1)/\lambda}
+#' When `y = 0`, the natural log transform is used.
+#'
+#' @references Box, G. E., & Cox, D. R. (1964). An analysis of transformations.
+#' Journal of the Royal Statistical Society. Series B (Methodological), 211-252.
+#' \url{https://www.jstor.org/stable/2984418}
 #'
 #' @export
 boxcox_trans <- function(p) {
+  if (abs(p) < 1e-07) return(log_trans())
+
+  trans <- function(x) {
+    if (any(x < 0)) {
+      stop("boxcox_trans must be given only positive values. Consider using modulus_trans instead?",
+        call. = F
+      )
+    }
+    (x^p - 1) / p
+  }
+
+  inv <- function(x) (x * p + 1)^(1 / p)
+
+  trans_new(
+    paste0("pow-", format(p)), trans, inv
+  )
+}
+
+#' @rdname boxcox_trans
+#' @details The modulus transformation implements a generalisation of the
```

Comment:
I think it's better to put all the details in the first doc block. That makes it easier to check that it reads well